datafusion/common/src/config.rs

```diff
@@ -397,6 +397,13 @@ impl ConfigOptions {
         Self::default()
     }
 
+    /// Creates a new [`ConfigOptions`] with extensions set to provided value
+    pub fn with_extensions(extensions: Extensions) -> Self {
```

Comment:
What would you think about making this a builder style API?

Like

```suggestion
    pub fn with_extensions(mut self, extensions: Extensions) -> Self {
```

So one could modify an existing ConfigOptions like:

```rust
let options = ConfigOptions::new()
  .with_extensions(extensions)
```

Reply:
The benefit of this style is that he naturally extends to setting other fields (rather than having to have a specialize constructor for each  field that we want to set)